src/components/Navbar.js

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useRef } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import { useKanban } from '../hooks/useKanban';
+
+export function Navbar() {
+  const { user, signInWithGoogle, signOut } = useAuth();
+  const { isDark, toggleTheme } = useTheme();
+  const { boardTitle, updateBoardTitle, addTask, updateColumn, columns } = useKanban();
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(boardTitle);
+  const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [showForms, setShowForms] = useState(false);
+  const titleInputRef = useRef(null);
+
+  const handleTitleClick = () => {
+    if (user) {
+      setIsEditingTitle(true);
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleChange = (e) => {
+const handleTitleBlur = async () => {
+const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);
+debouncedSetLoading(true);
+  setIsEditingTitle(false);
+  if (editedTitle.trim() && editedTitle !== boardTitle) {
+    try {
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
try {
      await updateBoardTitle(editedTitle.trim());
      // Use your preferred notification system (toast, alert, etc.)
      notifyUser({ message: 'Board title updated successfully', type: 'success' });
    } catch (error) {
      setEditedTitle(boardTitle);
      console.error('Failed to update board title:', error);
      notifyUser({ message: 'Failed to update board title', type: 'error' });
    }
```
The error handling in async operations (updateBoardTitle, addTask, updateColumn) lacks proper user feedback. Consider implementing a toast or notification system to inform users of success/failure.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;